pkg/deploy/stack_deployer.go

```diff
@@ -93,14 +93,48 @@ func NewLatticeServiceStackDeploy(cloud aws.Cloud, k8sClient client.Client, latt
 }
 
 func (d *latticeServiceStackDeployer) Deploy(ctx context.Context, stack core.Stack) error {
-	synthesizers := []ResourceSynthesizer{
-		lattice.NewTargetGroupSynthesizer(d.cloud, d.k8sclient, d.targetGroupManager, stack, d.latticeDataStore),
-		lattice.NewServiceSynthesizer(d.latticeServiceManager, stack, d.latticeDataStore),
-		lattice.NewTargetsSynthesizer(d.cloud, lattice.NewTargetsManager(d.cloud, d.latticeDataStore), stack, d.latticeDataStore),
-		lattice.NewListenerSynthesizer(d.listenerManager, stack, d.latticeDataStore),
-		lattice.NewRuleSynthesizer(d.ruleManager, stack, d.latticeDataStore),
+	targetGroupSynthesizer := lattice.NewTargetGroupSynthesizer(d.cloud, d.k8sclient, d.targetGroupManager, stack, d.latticeDataStore)
+	targetsSynthesizer := lattice.NewTargetsSynthesizer(d.cloud, lattice.NewTargetsManager(d.cloud, d.latticeDataStore), stack, d.latticeDataStore)
+	serviceSynthesizer := lattice.NewServiceSynthesizer(d.latticeServiceManager, stack, d.latticeDataStore)
+	listenerSynthesizer := lattice.NewListenerSynthesizer(d.listenerManager, stack, d.latticeDataStore)
+	ruleSynthesizer := lattice.NewRuleSynthesizer(d.ruleManager, stack, d.latticeDataStore)
+
+	//Handle targetGroups creation request
+	if err := targetGroupSynthesizer.Synthesize(ctx); err != nil {
+		return err
 	}
-	return deploy(ctx, stack, synthesizers)
+
+	//Handle targets "reconciliation" request (register intend-to-be-registered targets and deregister intend-to-be-registered targets)
+	if err := targetsSynthesizer.Synthesize(ctx); err != nil {
```

Comment:
I think it maybe  clear if you
if /* create path */ {
     t.SynthesizeTriggeredTargetGroupsCreation(...)
     create service(...)
}

Reply:
Got it, will change, although within the serviceSynthesizer.Synthesize() logic itself it could skip deletion deletion requests, but you are right, add a if else here make it more clear

Reply:
I found it's hard to do that, becuase for example, when hit the `targetGroupSynthesizer.SynthesizeTriggeredTargetGroupsCreation()` it need to handle **one list of ** TargetGroups creation or deletion requests (maybe can be a mix of creation and deletion for httproute mutation scenario?), it need to handle the the creation request and skip the deletion requests.

Similarly, for `targetGroupSynthesizer.SynthesizeTriggeredTargetGroupsDeletion()` it need to handle **ONE list of TargetGroups creation and deletion requests**, it need to handle the the deletion request and skip the deletion requests(`resTargetGroup.Spec.IsDeleted && !resTargetGroup.Spec.Config.IsServiceImport`).

Hard to do a if-else statement in `(d *latticeServiceStackDeployer) Deploy()`, because  `targetGroup.Spec.IsDeleted` flag exists for per targetgroup level

Reply:
I meant you can 
* still keep **one list** which  contains new TGs and delete TGs
* SynthesizeTriggeredTargetGroupsCreation() go through the list and takes care of `new TGs`
* SynthesizeTriggeredTargetGroupsDeletion() go through the list and takes care of `deleted TGs`

Reply:
NVM , saw your latest update, the change are fine.

Reply:
But, we need a unit test to cover func (d *latticeServiceStackDeployer) Deploy(ctx context.Context, stack core.Stack) error {).   Please create an github issue if you are `not` going to cover it in the PR